### DISCUSSION TITLE
Differs between local build vs deployed at Vercel

### DISCUSSION THREAD

**Discussion by ArenoRobert** [vercel/next.js#23173] [Help]
Hi guys!
My problem is regarding keeping persistent state using cookies and that it differs between my local build and deployed on Vercel.
Have a look at this test environment: https://non-prod.vercel.app/product/the-blue-one
Select Sweden in the Market Selector in the TopBar and then try to change the variants (Colors). 
Now you'll see that the price flashes between USD and SEK (USD is the default value for our cookies).

When I'm building the web locally (yarn build) I'm not receving this experience - only on Vercel.

What am I missing?

Robert

**Comment by akellbl4** [comment]
Hi Robert,
As I can see your page was statically pre-rendered in HTML. In your HTML prices are rendered in USD.
I can assume when the page is loaded with default prices in USD you read locale from a cookie and re-render page that prices.
It causes that flick.

**Reply by ArenoRobert** [reply]
Hi! Thanks,
I guess so as well. But how do I solve this?

**Reply by akellbl4** [reply]
I can suggest two options:
1. You can use [Next Internationalization](https://nextjs.org/docs/advanced-features/i18n-routing) to do so.
    While using i18n you can generate every page with prices in every currency.
    You will have navigation by locale `non-prod.vercel.app/[locale]/product/the-blue-one` and without locale for default locale.
2. Second option worse than the first one.
    You can render pages without prices but with a placeholder and then render prices only on the client-side.
